File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        t_index = None
        helper = 0

        while not t_index:
            if len(nums) == 1 and nums[0] == target:
                return 0

            middle = len(nums) // 2

            if middle == 0:
                return -1

            if target == nums[middle]:
                if helper == 0:
                    t_index = middle
                elif helper != 0:
                    t_index = helper + middle
                return t_index

            if target > nums[middle]:
                nums = nums[middle:]
                helper += middle

            else:
                nums = nums[:middle]


 

File: Data Structures & Algorithms/two-integer-sum/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const indices = new Map();

        for (const [i, n] of nums.entries()) {
            indices.set(n, i);
        };

        for (const [i, n] of nums.entries()) {
            const diff = target - n;
            if (indices.has(diff) && indices.get(diff) != i) {
                return [i, indices.get(diff)];
            };
        };        
    }
}
